### PR TITLE
Updating datacollector to use data serializer for Serialization

### PR DESCRIPTION
**Pull request by vagisha-nidhi**
Json data serialization fixed due to protocol version. Payload.ToObject changed to use JsonDataSerializer across all files.

### CODE REVIEW

File: test/Microsoft.TestPlatform.CommunicationUtilities.UnitTests/DataCollectionRequestHandlerTests.cs [hunk 1]
```diff
@@ -35,7 +35,7 @@ public DataCollectionRequestHandlerTests()
             this.mockDataCollectionManager = new Mock<IDataCollectionManager>();
             this.mockDataCollectionTestCaseEventHandler = new Mock<IDataCollectionTestCaseEventHandler>();
             this.mockDataCollectionTestCaseEventHandler.Setup(x => x.WaitForRequestHandlerConnection(It.IsAny<int>())).Returns(true);
-            this.requestHandler = new TestableDataCollectionRequestHandler(this.mockCommunicationManager.Object, this.mockMessageSink.Object, this.mockDataCollectionManager.Object, this.mockDataCollectionTestCaseEventHandler.Object);
+            this.requestHandler = new TestableDataCollectionRequestHandler(this.mockCommunicationManager.Object, this.mockMessageSink.Object, this.mockDataCollectionManager.Object, this.mockDataCollectionTestCaseEventHandler.Object, JsonDataSerializer.Instance);
```
Comment (harshjain2) [inline test/Microsoft.TestPlatform.CommunicationUtilities.UnitTests/DataCollectionRequestHandlerTests.cs]: Can we use mock here?

File: test/TestAssets/OutOfProcDataCollector/SampleDataCollector.cs [hunk 2]
```diff
@@ -42,9 +42,11 @@ private void Events_TestCaseEnd(object sender, TestCaseEndEventArgs e)
         private void Events_TestCaseStart(object sender, TestCaseStartEventArgs e)
         {
             this.logger.LogWarning(this.context.SessionDataCollectionContext, "TestCaseStarted " + e.TestCaseName);
+            this.logger.LogWarning(this.context.SessionDataCollectionContext, "TestCaseStarted " + e.TestElement.FullyQualifiedName);
             var filename = Path.Combine(AppContext.BaseDirectory, "testcasefilename" + i++ + ".txt");
             File.WriteAllText(filename, string.Empty);
             this.dataCollectionSink.SendFileAsync(e.Context, filename, true);
+
```
Comment (harshjain2) [inline test/TestAssets/OutOfProcDataCollector/SampleDataCollector.cs]: nit : remote extra line.

File: src/Microsoft.TestPlatform.CoreUtilities/Helpers/FileHelper.cs [hunk 3]
```diff
@@ -70,5 +70,11 @@ public void MoveFile(string sourcePath, string destinationPath)
         {
             File.Move(sourcePath, destinationPath);
         }
+
+        /// <inheritdoc/>
+        public void AppendAllText(string sourcePath, string contents)
```
Comment (singhsarab) [inline src/Microsoft.TestPlatform.CoreUtilities/Helpers/FileHelper.cs]: These changes seem unrelated to this fix.

**Review by singhsarab — APPROVED** [review]
Approving with some suggestions.
Wait for @harshjain2 to approve.

File: test/Microsoft.TestPlatform.AcceptanceTests/DataCollectionTests.cs [hunk 4]
```diff
@@ -113,6 +113,8 @@ private void VaildateDataCollectorOutput()
             this.StdOutputContains("filename.txt");
             this.StdOutputContains("TestCaseStarted");
             this.StdOutputContains("TestCaseEnded");
+            this.StdOutputContains("SampleUnitTestProject2.UnitTest1.PassingTest2");
```
Comment (harshjain2) [inline test/Microsoft.TestPlatform.AcceptanceTests/DataCollectionTests.cs]: 👍

**Review by harshjain2 — APPROVED** [review]

**Pull request merged**